VCDA-557 & VCDA-607: Basic NFS PV support

sahithi opened the pull request:
1. Modify Ubuntu template script to install NFS-client
2. Enhance add-node code to create a node of NFS server (covers installation of nfs server)

hodgesrm commented:
Reviewed 1 of 4 files at r1, 1 of 2 files at r2, 1 of 1 files at r3, 3 of 3 files at r4.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---

*[a discussion](https://reviewable.io/reviews/vmware/container-service-extension/78#-LB73zZa9ddPo65-WMiB:-LB73zZa9ddPo65-WMiC:bdgktgw) (no related file):*
What documentation is provided for managing NFS once the node is up and running? 

---

*[container_service_extension/broker.py, line 612 at r4](https://reviewable.io/reviews/vmware/container-service-extension/78#-LB71vqZ6oPg32vm64wU:-LB71vqZ6oPg32vm64wV:b-8iq9sx) ([raw file](https://github.com/vmware/container-service-extension/blob/96aacb3606f030cd6084eb956f6760e0cba462c1/container_service_extension/broker.py#L612)):*
> ```Python
>                              self.cluster_name,
>                              self.cluster_id))
>             if self.body['node_type'] == TYPE_WORKER:
> ```

Nit: This would be clearer as an elif: since self.body['node_type'] can have only one value. 

---

*[container_service_extension/client/cse.py, line 361 at r4](https://reviewable.io/reviews/vmware/container-service-extension/78#-LB73dVY4tvYZuOxHGUM:-LB73dVY4tvYZuOxHGUN:b-d4l49z) ([raw file](https://github.com/vmware/container-service-extension/blob/96aacb3606f030cd6084eb956f6760e0cba462c1/container_service_extension/client/cse.py#L361)):*
> ```Python
>     required=False,
>     default='node',
>     type=click.Choice(['node', 'nfs']),
> ```

It's a CLI interface change but might it not make more sense to change 'node' to 'worker'?  Will other node types be added later?

---


*Comments from [Reviewable](https://reviewable.io/reviews/vmware/container-service-extension/78)*
<!-- Sent from Reviewable.io -->

sahithi commented:
Review status: all files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[a discussion](https://reviewable.io/reviews/vmware/container-service-extension/78#-LB73zZa9ddPo65-WMiB:-LB8Hev83LhFqMt4KePp:b-9rjkpu) (no related file):*
<details><summary><i>Previously, hodgesrm (Robert Hodges) wrote…</i></summary><blockquote>

What documentation is provided for managing NFS once the node is up and running? 
</blockquote></details>

Robert, I have Jira tasks lined up for documentation and for a way to provide NFS details as part of Cluster Info/Node info commands.

The next immediate goal is to provide a way for users to know IP & other details of the NFS server.

Initially I was planning to output the IP of NFS server for "vcd cse node add --type nfs" command but I realized its a bit trickier with the current design and I will soon consult you and Paco about it.

---

*[container_service_extension/broker.py, line 612 at r4](https://reviewable.io/reviews/vmware/container-service-extension/78#-LB71vqZ6oPg32vm64wU:-LB8IxQb-p1Wn-tfGNFY:b-896fix) ([raw file](https://github.com/vmware/container-service-extension/blob/96aacb3606f030cd6084eb956f6760e0cba462c1/container_service_extension/broker.py#L612)):*
<details><summary><i>Previously, hodgesrm (Robert Hodges) wrote…</i></summary><blockquote>

Nit: This would be clearer as an elif: since self.body['node_type'] can have only one value. 
</blockquote></details>

Done.

---

*[container_service_extension/client/cse.py, line 361 at r4](https://reviewable.io/reviews/vmware/container-service-extension/78#-LB73dVY4tvYZuOxHGUM:-LB8IzOA1FmV4u-fMe0a:bmspy1s) ([raw file](https://github.com/vmware/container-service-extension/blob/96aacb3606f030cd6084eb956f6760e0cba462c1/container_service_extension/client/cse.py#L361)):*
<details><summary><i>Previously, hodgesrm (Robert Hodges) wrote…</i></summary><blockquote>

It's a CLI interface change but might it not make more sense to change 'node' to 'worker'?  Will other node types be added later?
</blockquote></details>

Agreed.  To keep it consistent across the project, we may have change the current naming convention of worker nodes then

Current naming convention in the cluster looks like below:
mstr-3654
node-8976
node-6578
nfsd-6932

I can replace node -> wrkr. However, this may cause slight inconsistency in the existing clusters, when user tries to add new nodes. But I think "wrkr" sounds better than "node"

Paco, are you okay with it? @pacogomez 

---


*Comments from [Reviewable](https://reviewable.io/reviews/vmware/container-service-extension/78)*
<!-- Sent from Reviewable.io -->

pacogomez commented:
Review status: all files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[container_service_extension/broker.py, line 612 at r4](https://reviewable.io/reviews/vmware/container-service-extension/78#-LB71vqZ6oPg32vm64wU:-LB8Rbxf0WbNHIyYlGWF:b-jaje5o) ([raw file](https://github.com/vmware/container-service-extension/blob/96aacb3606f030cd6084eb956f6760e0cba462c1/container_service_extension/broker.py#L612)):*
<details><summary><i>Previously, sahithi (Sahithi Ayloo) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

default node count for node type nfsd should be 1

---

*[container_service_extension/cluster.py, line 202 at r4](https://reviewable.io/reviews/vmware/container-service-extension/78#-LB8RHH-ALDpuTAkXDkA:-LB8RHH-ALDpuTAkXDkB:b-373x5g) ([raw file](https://github.com/vmware/container-service-extension/blob/96aacb3606f030cd6084eb956f6760e0cba462c1/container_service_extension/cluster.py#L202)):*
> ```Python
>             execute_script_in_nodes(config, vapp,
>                                     template['admin_password'],
>                                     script, nodes)
> ```

add another debug line to indicate the customization is finished

---

*[container_service_extension/client/cse.py, line 361 at r4](https://reviewable.io/reviews/vmware/container-service-extension/78#-LB73dVY4tvYZuOxHGUM:-LB8Nt9H7FcPmTUK81_q:bqjnz20) ([raw file](https://github.com/vmware/container-service-extension/blob/96aacb3606f030cd6084eb956f6760e0cba462c1/container_service_extension/client/cse.py#L361)):*
<details><summary><i>Previously, sahithi (Sahithi Ayloo) wrote…</i></summary><blockquote>

Agreed.  To keep it consistent across the project, we may have change the current naming convention of worker nodes then

Current naming convention in the cluster looks like below:
mstr-3654
node-8976
node-6578
nfsd-6932

I can replace node -> wrkr. However, this may cause slight inconsistency in the existing clusters, when user tries to add new nodes. But I think "wrkr" sounds better than "node"

Paco, are you okay with it? @pacogomez 
</blockquote></details>

Initially in Kubernetes, the name 'worker' was used to identify the worker nodes, but later it was changed to just 'node'. I followed that pattern in CSE at the beginning and then switched to just call 'node' to the worker nodes, and 'master' to the master nodes. I would prefer if we stay that way to follow the Kubernetes nomenclature, even if a node of type node seems a bit obvious or self descriptive. I would prefer if we keep the type name 'node' for worker nodes both in the source code and in the CLI commands. That would be my preference.

---

*[scripts/cust-ubuntu-16.04.sh, line 56 at r4](https://reviewable.io/reviews/vmware/container-service-extension/78#-LB8PzVcDWWSJTv0rf_q:-LB8PzVcDWWSJTv0rf_r:b-lc55ul) ([raw file](https://github.com/vmware/container-service-extension/blob/96aacb3606f030cd6084eb956f6760e0cba462c1/scripts/cust-ubuntu-16.04.sh#L56)):*
> ```Shell
> 
> echo 'installing nfs-client'
> apt-get -q install -y nfs-common
> ```

during customization of the master (general) template, all the required software should be pre-installed in the VM, even if the VM would later be instantiate as a worker node. That means to install here the nfs server components.

---

*[scripts/nfs-ubuntu-16.04.sh, line 3 at r4](https://reviewable.io/reviews/vmware/container-service-extension/78#-LB8QYWxAi_IHJb3r_GD:-LB8QYWxAi_IHJb3r_GE:b-5ja0m0) ([raw file](https://github.com/vmware/container-service-extension/blob/96aacb3606f030cd6084eb956f6760e0cba462c1/scripts/nfs-ubuntu-16.04.sh#L3)):*
> ```Shell
> #!/usr/bin/env bash
> set -e
> apt-get -q install -y nfs-kernel-server
> ```

during instantiation of the master template as a customer node (nfsd) no installation of software should be done, since the VM might not be connected to the internet. At this stage, only configuration commands should be executed in this script. The script might just be empty and have it as a placeholder for future customization needs. Please move this line to the cust-*.sh script

---


*Comments from [Reviewable](https://reviewable.io/reviews/vmware/container-service-extension/78)*
<!-- Sent from Reviewable.io -->

pacogomez commented:
Review status: all files reviewed at latest revision, 6 unresolved discussions, some commit checks failed.

---

*[a discussion](https://reviewable.io/reviews/vmware/container-service-extension/78#-LB73zZa9ddPo65-WMiB:-LB8Rv-x0OZOFQHUK4k3:b-5az24a) (no related file):*
<details><summary><i>Previously, sahithi (Sahithi Ayloo) wrote…</i></summary><blockquote>

Robert, I have Jira tasks lined up for documentation and for a way to provide NFS details as part of Cluster Info/Node info commands.

The next immediate goal is to provide a way for users to know IP & other details of the NFS server.

Initially I was planning to output the IP of NFS server for "vcd cse node add --type nfs" command but I realized its a bit trickier with the current design and I will soon consult you and Paco about it.
</blockquote></details>

yes, lets get this finished with current functionality and will add improvements later. see my comments below. great job!

---


*Comments from [Reviewable](https://reviewable.io/reviews/vmware/container-service-extension/78)*
<!-- Sent from Reviewable.io -->

pacogomez commented:
Review status: all files reviewed at latest revision, 6 unresolved discussions, some commit checks failed.

---

*[scripts/nfs-ubuntu-16.04.sh, line 3 at r4](https://reviewable.io/reviews/vmware/container-service-extension/78#-LB8QYWxAi_IHJb3r_GD:-LB8USRc7iKyVqa5iDUi:b-y43vqx) ([raw file](https://github.com/vmware/container-service-extension/blob/96aacb3606f030cd6084eb956f6760e0cba462c1/scripts/nfs-ubuntu-16.04.sh#L3)):*
<details><summary><i>Previously, pacogomez (Paco Gómez) wrote…</i></summary><blockquote>

during instantiation of the master template as a customer node (nfsd) no installation of software should be done, since the VM might not be connected to the internet. At this stage, only configuration commands should be executed in this script. The script might just be empty and have it as a placeholder for future customization needs. Please move this line to the cust-*.sh script
</blockquote></details>

if the file is empty, it could just be removed. if any customization is needed here, the file needs to be included in the `setup.cfg` file to be included in the package.

---


*Comments from [Reviewable](https://reviewable.io/reviews/vmware/container-service-extension/78)*
<!-- Sent from Reviewable.io -->

pacogomez commented:
Review status: all files reviewed at latest revision, 6 unresolved discussions, some commit checks failed.

---

*[container_service_extension/cluster.py, line 199 at r4](https://reviewable.io/reviews/vmware/container-service-extension/78#-LBCbXTXBpJo_FtLanng:-LBCbXTXBpJo_FtLannh:bo0dfr3) ([raw file](https://github.com/vmware/container-service-extension/blob/96aacb3606f030cd6084eb956f6760e0cba462c1/container_service_extension/cluster.py#L199)):*
> ```Python
>             LOGGER.debug('Installing NFS server on %s' %
>                          spec['target_vm_name'])
>             script = get_data_file('nfs-%s.sh' % template['name'])
> ```

travis ci build complains about this line

---


*Comments from [Reviewable](https://reviewable.io/reviews/vmware/container-service-extension/78)*
<!-- Sent from Reviewable.io -->

hodgesrm commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: all files reviewed at latest revision, 7 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io/reviews/vmware/container-service-extension/78#-:-LBNecy283nyGGFHonlj:bnfp4nl)*
<!-- Sent from Reviewable.io -->

pacogomez commented:
Review status: 1 of 6 files reviewed at latest revision, 7 unresolved discussions, some commit checks failed.

---

*[scripts/cust-ubuntu-16.04.sh, line 57 at r5](https://reviewable.io/reviews/vmware/container-service-extension/78#-LBORYHz0-98Ch2mTQMw:-LBORYI-4oHENucR_4V7:b-2o0tua) ([raw file](https://github.com/vmware/container-service-extension/blob/025c07f3f0fe6f2bf46c38153f27c0c83d469124/scripts/cust-ubuntu-16.04.sh#L57)):*
> ```Shell
> echo 'installing required software for NFS'
> apt-get -q install -y nfs-common
> apt-get -q install -y nfs-kernel-server
> ```

does nfs-kernel-server include nfs-common? if yes, installation of nfs-common is redundant and should not be included.

---


*Comments from [Reviewable](https://reviewable.io/reviews/vmware/container-service-extension/78)*
<!-- Sent from Reviewable.io -->

sahithi commented:
Review status: 1 of 6 files reviewed at latest revision, 8 unresolved discussions, some commit checks failed.

---

*[container_service_extension/broker.py, line 612 at r4](https://reviewable.io/reviews/vmware/container-service-extension/78#-LB71vqZ6oPg32vm64wU:-LBOQ6itAiShWqqlebau:bfr2aj) ([raw file](https://github.com/vmware/container-service-extension/blob/96aacb3606f030cd6084eb956f6760e0cba462c1/container_service_extension/broker.py#L612)):*
<details><summary><i>Previously, pacogomez (Paco Gómez) wrote…</i></summary><blockquote>

default node count for node type nfsd should be 1
</blockquote></details>

Currently Click framework assigns default node count=2 for "create node" command. Can we keep it the same for NFS as it uses the same command?

---

*[container_service_extension/cluster.py, line 199 at r4](https://reviewable.io/reviews/vmware/container-service-extension/78#-LBCbXTXBpJo_FtLanng:-LBORAJt1mdIj0Dsh0Qq:b-it303p) ([raw file](https://github.com/vmware/container-service-extension/blob/96aacb3606f030cd6084eb956f6760e0cba462c1/container_service_extension/cluster.py#L199)):*
<details><summary><i>Previously, pacogomez (Paco Gómez) wrote…</i></summary><blockquote>

travis ci build complains about this line
</blockquote></details>

I missed committing the fix for it. It is all good now.

---

*[container_service_extension/cluster.py, line 202 at r4](https://reviewable.io/reviews/vmware/container-service-extension/78#-LB8RHH-ALDpuTAkXDkA:-LBORpdn6czaGQEdCXMk:b8lxpgc) ([raw file](https://github.com/vmware/container-service-extension/blob/96aacb3606f030cd6084eb956f6760e0cba462c1/container_service_extension/cluster.py#L202)):*
<details><summary><i>Previously, pacogomez (Paco Gómez) wrote…</i></summary><blockquote>

add another debug line to indicate the customization is finished
</blockquote></details>

I moved installation to customization script. The current script only enables nfs-server. Added the log line indicating the same

---

*[container_service_extension/client/cse.py, line 361 at r4](https://reviewable.io/reviews/vmware/container-service-extension/78#-LB73dVY4tvYZuOxHGUM:-LBOS6hQ8YpbACwdjkiY:b-6tf1vw) ([raw file](https://github.com/vmware/container-service-extension/blob/96aacb3606f030cd6084eb956f6760e0cba462c1/container_service_extension/client/cse.py#L361)):*
<details><summary><i>Previously, pacogomez (Paco Gómez) wrote…</i></summary><blockquote>

Initially in Kubernetes, the name 'worker' was used to identify the worker nodes, but later it was changed to just 'node'. I followed that pattern in CSE at the beginning and then switched to just call 'node' to the worker nodes, and 'master' to the master nodes. I would prefer if we stay that way to follow the Kubernetes nomenclature, even if a node of type node seems a bit obvious or self descriptive. I would prefer if we keep the type name 'node' for worker nodes both in the source code and in the CLI commands. That would be my preference.
</blockquote></details>

It makes sense to keep naming convention aligned with Kubernetes.

---

*[scripts/cust-ubuntu-16.04.sh, line 56 at r4](https://reviewable.io/reviews/vmware/container-service-extension/78#-LB8PzVcDWWSJTv0rf_q:-LBOSHII6SGK09N1EpRn:b-896fix) ([raw file](https://github.com/vmware/container-service-extension/blob/96aacb3606f030cd6084eb956f6760e0cba462c1/scripts/cust-ubuntu-16.04.sh#L56)):*
<details><summary><i>Previously, pacogomez (Paco Gómez) wrote…</i></summary><blockquote>

during customization of the master (general) template, all the required software should be pre-installed in the VM, even if the VM would later be instantiate as a worker node. That means to install here the nfs server components.
</blockquote></details>

Done.

---

*[scripts/cust-ubuntu-16.04.sh, line 57 at r5](https://reviewable.io/reviews/vmware/container-service-extension/78#-LBORYHz0-98Ch2mTQMw:-LBOSHlxEQkVjFhlAkAA:b-7y7th1) ([raw file](https://github.com/vmware/container-service-extension/blob/025c07f3f0fe6f2bf46c38153f27c0c83d469124/scripts/cust-ubuntu-16.04.sh#L57)):*
<details><summary><i>Previously, pacogomez (Paco Gómez) wrote…</i></summary><blockquote>

does nfs-kernel-server include nfs-common? if yes, installation of nfs-common is redundant and should not be included.
</blockquote></details>

It does not include nfs-common. 

---

*[scripts/nfs-ubuntu-16.04.sh, line 3 at r4](https://reviewable.io/reviews/vmware/container-service-extension/78#-LB8QYWxAi_IHJb3r_GD:-LBOSdRf4iarOP9p2VXT:b-896fix) ([raw file](https://github.com/vmware/container-service-extension/blob/96aacb3606f030cd6084eb956f6760e0cba462c1/scripts/nfs-ubuntu-16.04.sh#L3)):*
<details><summary><i>Previously, pacogomez (Paco Gómez) wrote…</i></summary><blockquote>

if the file is empty, it could just be removed. if any customization is needed here, the file needs to be included in the `setup.cfg` file to be included in the package.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io/reviews/vmware/container-service-extension/78)*
<!-- Sent from Reviewable.io -->

pacogomez commented:
Review status: 1 of 7 files reviewed at latest revision, 8 unresolved discussions.

---

*[scripts/cust-ubuntu-16.04.sh, line 57 at r5](https://reviewable.io/reviews/vmware/container-service-extension/78#-LBORYHz0-98Ch2mTQMw:-LBOTBQFExQAisAw58DY:b-u6tlru) ([raw file](https://github.com/vmware/container-service-extension/blob/025c07f3f0fe6f2bf46c38153f27c0c83d469124/scripts/cust-ubuntu-16.04.sh#L57)):*
<details><summary><i>Previously, sahithi (Sahithi Ayloo) wrote…</i></summary><blockquote>

It does not include nfs-common. 
</blockquote></details>

ok, let's at least install both packages in one single line

---

*[scripts/nfs-ubuntu-16.04.sh, line 3 at r4](https://reviewable.io/reviews/vmware/container-service-extension/78#-LB8QYWxAi_IHJb3r_GD:-LBOTH0U2k8slC0CFBP0:b-ong1yq) ([raw file](https://github.com/vmware/container-service-extension/blob/96aacb3606f030cd6084eb956f6760e0cba462c1/scripts/nfs-ubuntu-16.04.sh#L3)):*
<details><summary><i>Previously, sahithi (Sahithi Ayloo) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

file should be `nfsd-ubuntu-16.04` to match the name of the node type.

---


*Comments from [Reviewable](https://reviewable.io/reviews/vmware/container-service-extension/78)*
<!-- Sent from Reviewable.io -->

sahithi commented:
Review status: 1 of 7 files reviewed at latest revision, 8 unresolved discussions.

---

*[container_service_extension/cluster.py, line 199 at r4](https://reviewable.io/reviews/vmware/container-service-extension/78#-LBCbXTXBpJo_FtLanng:-LBOkEPcAWbdrihH3AhL:b5awexu) ([raw file](https://github.com/vmware/container-service-extension/blob/96aacb3606f030cd6084eb956f6760e0cba462c1/container_service_extension/cluster.py#L199)):*
<details><summary><i>Previously, pacogomez (Paco Gómez) wrote…</i></summary><blockquote>

the import statement should go to at the top of the file, unless there is a circular dependency
</blockquote></details>

I was under the same impression. However as this import is for a specific method, it is not allowing me to put at the top. The initial travis build was failing for the same reason.

---

*[container_service_extension/client/cse.py, line 361 at r4](https://reviewable.io/reviews/vmware/container-service-extension/78#-LB73dVY4tvYZuOxHGUM:-LBOk9y--M2J9gDNS8DG:b-896fix) ([raw file](https://github.com/vmware/container-service-extension/blob/96aacb3606f030cd6084eb956f6760e0cba462c1/container_service_extension/client/cse.py#L361)):*
<details><summary><i>Previously, sahithi (Sahithi Ayloo) wrote…</i></summary><blockquote>

It makes sense to keep naming convention aligned with Kubernetes.
</blockquote></details>

Done.

---

*[scripts/cust-ubuntu-16.04.sh, line 57 at r5](https://reviewable.io/reviews/vmware/container-service-extension/78#-LBORYHz0-98Ch2mTQMw:-LBOk7m39v4IC-IXsXLi:b-896fix) ([raw file](https://github.com/vmware/container-service-extension/blob/025c07f3f0fe6f2bf46c38153f27c0c83d469124/scripts/cust-ubuntu-16.04.sh#L57)):*
<details><summary><i>Previously, pacogomez (Paco Gómez) wrote…</i></summary><blockquote>

ok, let's at least install both packages in one single line
</blockquote></details>

Done.

---

*[scripts/nfs-ubuntu-16.04.sh, line 3 at r4](https://reviewable.io/reviews/vmware/container-service-extension/78#-LB8QYWxAi_IHJb3r_GD:-LBOjVdgD6J2mv07UxCo:b-896fix) ([raw file](https://github.com/vmware/container-service-extension/blob/96aacb3606f030cd6084eb956f6760e0cba462c1/scripts/nfs-ubuntu-16.04.sh#L3)):*
<details><summary><i>Previously, pacogomez (Paco Gómez) wrote…</i></summary><blockquote>

file should be `nfsd-ubuntu-16.04` to match the name of the node type.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io/reviews/vmware/container-service-extension/78)*
<!-- Sent from Reviewable.io -->

pacogomez commented:
Review status: 0 of 7 files reviewed at latest revision, 7 unresolved discussions.

---

*[a discussion](https://reviewable.io/reviews/vmware/container-service-extension/78#-LB73zZa9ddPo65-WMiB:-LBWd4a23pl5-Zc-Xvd-:b-2atd4u) (no related file):*
<details><summary><i>Previously, pacogomez (Paco Gómez) wrote…</i></summary><blockquote>

yes, lets get this finished with current functionality and will add improvements later. see my comments below. great job!
</blockquote></details>

LGTM, the only suggestion is to have the default number of nodes in `node create` command to 1

---

*[container_service_extension/broker.py, line 612 at r4](https://reviewable.io/reviews/vmware/container-service-extension/78#-LB71vqZ6oPg32vm64wU:-LBWcKMH0YRhAtTnh98O:bekimdm) ([raw file](https://github.com/vmware/container-service-extension/blob/96aacb3606f030cd6084eb956f6760e0cba462c1/container_service_extension/broker.py#L612)):*
<details><summary><i>Previously, sahithi (Sahithi Ayloo) wrote…</i></summary><blockquote>

Currently Click framework assigns default node count=2 for "create node" command. Can we keep it the same for NFS as it uses the same command?
</blockquote></details>

it might be better to have default node count = 1 in the create node command

---


*Comments from [Reviewable](https://reviewable.io/reviews/vmware/container-service-extension/78)*
<!-- Sent from Reviewable.io -->

pacogomez commented:
Reviewed 1 of 6 files at r5, 6 of 7 files at r7.
Review status: all files reviewed at latest revision, 7 unresolved discussions.

---

*[a discussion](https://reviewable.io/reviews/vmware/container-service-extension/78#-LB73zZa9ddPo65-WMiB:-LBWdLMxED-loYn1t2Yf:bb6ywmx) (no related file):*
<details><summary><i>Previously, pacogomez (Paco Gómez) wrote…</i></summary><blockquote>

LGTM, the only suggestion is to have the default number of nodes in `node create` command to 1
</blockquote></details>

LGTM

---


*Comments from [Reviewable](https://reviewable.io/reviews/vmware/container-service-extension/78)*
<!-- Sent from Reviewable.io -->

pacogomez commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: all files reviewed at latest revision, 7 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io/reviews/vmware/container-service-extension/78#-:-LBWdZxP0y7eDxzjnc_3:bnfp4nl)*
<!-- Sent from Reviewable.io -->

pacogomez commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 1 of 7 files at r7, 1 of 1 files at r8.
Review status: all files reviewed at latest revision, 7 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io/reviews/vmware/container-service-extension/78#-:-LBY257l9DxnpvNolIt1:bnfp4nl)*
<!-- Sent from Reviewable.io -->
